CONTRIBUTING.md

```diff
@@ -1,77 +1,175 @@
+**Phase 3: 3rd Party Devs Contribute & Run Tests**
+Contributors will be able to submit PRs and run tests locally. We’re working to untangle private dependencies and make test infrastructure publicly accessible.
+**Phase 4: GitHub as Center of Gravity**
+GitHub becomes the primary place for development, issue tracking, and community engagement. Internal mirrors will be phased out.
+
+### [WIP] New Contributors
+Contributions from the community are greatly appreciated. We mark the most straightforward issues with labels. These issues are the best place to start if you are interested in contributing but are new to the codebase.
+
+- [good first issues](https://github.com/Microsoft/microsoft-ui-xaml/labels/good first issue)
+- [help wanted](https://github.com/orgs/microsoft/projects/1868/views/12)
 
+Another great way to help is by up-voting and commenting on feature proposals:
+- [feature request](https://github.com/Microsoft/microsoft-ui-xaml/labels/feature request)
+
+### [WIP] Code Contribution Process
+We use and recommend the following workflow:
+1. Create an issue for your work. 
+    - You can skip this step for trivial changes or if there is an existing issue for the bug/feature.
+    - If your change adds or changes public APIs or UI, first follow the [New Feature or API Process](docs/feature_proposal_process.md).
+    - Clearly state that you are going to take on implementing it, if that's the case. You can request that the issue be assigned to you. Note: The issue filer and the implementer don't have to be the same person.
+1. Create a personal fork of the repository on GitHub (if you don't already have one).
+1. Create a branch off of main (git checkout -b mybranch). 
+    - Name the branch so that it clearly communicates your intentions (i.e. /user/janedoe/fix-datepicker).
+    - Branches are useful since they isolate your changes from incoming changes from upstream. They also enable you to create multiple PRs from the same fork.
+1. Make and commit your changes. 
+    - Please follow our [Commit Messages](#commit-messages) guidance.
+1. Add new tests corresponding to your change, if applicable.
+1. Build the repository with your changes. 
+    - Make sure that the builds are clean.
+    - Make sure that the tests are all passing, including your new tests.
+1. Create a pull request (PR) against this repository's main branch. 
+    - Push your changes to your fork on GitHub (if you haven't already).
+    - Note: It is okay for your PR to include a large number of commits. Once your change is accepted, you will be asked to squash your commits into one or some appropriately small number of commits before your PR is merged.
+    - Note: It is okay to create your PR as "[WIP]" before the implementation is done. This can be useful if you'd like to start the feedback process while you're still working on the implementation. State that this is the case in the initial PR comment.
```

Comment:
```suggestion
    - Note: It is okay to create your PR as a draft PR before the implementation is done. This can be useful if you'd like to start the feedback process while you're still working on the implementation. When you're ready for deeper consideration, change the PR's state to ready to review (not draft).
```